server/game/cards/03_TWI/units/501stLiberator.ts

```diff
@@ -0,0 +1,25 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { NonLeaderUnitCard } from '../../../core/card/NonLeaderUnitCard';
+import { Trait } from '../../../core/Constants';
+
+export default class _501stLiberator extends NonLeaderUnitCard {
+    protected override getImplementationId () {
+        return {
+            id: '7252148824',
+            internalName: '501st-liberator',
+        };
+    }
+
+    public override setupCardAbilities () {
+        this.addWhenPlayedAbility({
+            title: 'If you control another Republic unit, you may heal 3 damage from a base.',
+            immediateEffect: AbilityHelper.immediateEffects.conditional({
```

Comment:
The text says "heal 3 damage from a base" which means the player should be allowed to choose either base. So this should be a target resolver to allow the player to choose. The `immediateEffect` inside the `targetResolver` would be what you currently have here. Tests would need to be updated obviously